packages/api/src/store/queue.ts

```diff
@@ -0,0 +1,193 @@
+import sql, { SQLStatement } from "sql-template-strings";
+import { DB } from "./db";
+import logger from "../logger";
+import { BadRequestError } from "./errors";
+import { QueryResult, PoolClient, Pool } from "pg";
+import Table from "./table";
+import { Queue } from "../schema/types";
+import { Notification } from "pg/index";
+import { TableSchema, GetOptions, DBObject, DBLegacyObject } from "./types";
+
+const DEFAULT_SORT = "id ASC";
+
+export default class QueueTable extends Table<Queue> {
+  db: DB;
+  schema: TableSchema;
+  client: PoolClient;
+  listener: PoolClient;
+  onNotification: any;
+  channel = "webhook_main";
+  constructor({ db, schema }) {
+    super({ db, schema });
+    // this.start();
+  }
+
+  async start() {
+    // this.client = await this.db.pool.connect();
+    await this.listen();
+  }
+
+  async stop() {
+    try {
+      this.listener.removeAllListeners("error");
+      this.listener.removeAllListeners("notification");
+      this.listener.removeAllListeners("end");
+      this.listener.removeAllListeners("notice");
+      this.listener.release();
+      // if (this.client) {
+      //   this.client.release();
+      // }
+    } catch (error) {
+      console.log("error releasing pg client", error);
+    }
+  }
+
+  unsetMsgHandler() {
+    this.setMsgHandler(QueueTable.prototype.handleMsg);
+  }
+
+  async listen() {
+    this.listener = await this.db.pool.connect();
+    await this.listener.query(`LISTEN ${this.channel}`);
+    this.onNotification = this.listener.on("notification", this.handleMsg);
+
+    this.listener.on("error", (error) => {
+      console.error("Msg Queue Listener Error ", error);
+    });
+    this.listener.on("end", () => {
+      console.log("Msq queue listener ended");
+    });
+    this.listener.on("notice", (error) => {
+      console.warn("Msg Queue Listener Notice: ", error);
+    });
+  }
+
+  async setMsgHandler(
+    func: (msg: Notification) => Promise<void>
+  ): Promise<any> {
+    console.log("setting msg handler");
+    // this.handleMsg = func;
+    this.listener.removeListener("notification", this.handleMsg);
+    this.listener.on("notification", func);
+    return true;
+  }
+
+  async handleMsg(msg: Notification) {
+    // emit it here
+    console.log("got NOTIFICATION: ", msg.channel, msg.payload);
+  }
+  // name: string;
+  // constructor({ db, schema }) {
+  //   super({ db, schema });
+  //   this.db = db;
+  //   this.schema = schema;
+  //   this.name = schema.queue;
+  // }
+
+  // get next event in queue
+  async pop(processFunc: (q: Queue) => any): Promise<Queue> {
+    let res: QueryResult<DBLegacyObject>;
+    this.client = await this.db.pool.connect();
```

Comment:
It shouldn't be assigned to the object field, `client` should be local variable.